libvast/src/option_declaration_set.cpp

```diff
@@ -124,32 +124,38 @@ std::string option_declaration_set::usage() const {
     auto& shorts = x.short_names();
     arg << "  "; 
     if (!shorts.empty()) {
-      arg << "-" <<shorts[0];
-      arg << " [";
+      auto i = shorts.begin();
+      auto e = shorts.end();
+      arg << "-" << *i << " [";
+      for (++i; i != e; ++i)
+        arg << "-" << *i << ", ";
+      arg << "--" << x.long_name() << ']';
+    } else {
+      arg << "--" << x.long_name();
     }
-    if (shorts.size() > 1)
-      for (auto i = 1u; i < shorts.size(); ++i)
-        arg << "-" << shorts[i] << ",";
-    arg << "--" << x.long_name();
-    if (!shorts.empty())
-      arg << "]";
     if (x.has_argument())
       arg << " arg";
     return arg.str();
   };
   // Calculate the max size the argument column
-  std::vector<size_t> arg_sizes;
-  std::transform(
-    long_opts_.begin(), long_opts_.end(), std::back_inserter(arg_sizes),
-    [&](const auto& x) { return build_argument(*x.second).size(); });
-  auto help_column_width = *max_element(arg_sizes.begin(), arg_sizes.end());
+  std::vector<std::string> args;
+  args.reserve(size());
+  for (auto& x: long_opts_)
+    args.emplace_back(build_argument(*x.second));
+  size_t help_column_width = 0;
+  if (auto it = max_element(args.begin(), args.end(),
```

Comment:
I see two issues with this. First, the lambda copies strings around (should've `auto& a, auto& b`). Second, we aren't at all interested in the position of the `max_element`.

This is a simple fold operation. Unfortunately, C++ uses the misguiding name `accumulate` for this algorithm.

I'd rewrite it somewhat like this:

```cpp
    auto max_str_size = [](size_t a, auto& b) { return std::max(a, b.size()); }
    auto max_column_width = std::accumulate(args.begin(), args.end(), size_t{0}, max_str_size);
```